src/plugins/console/public/lib/mappings/mappings.ts

```diff
@@ -307,53 +358,52 @@ export function clearSubscriptions() {
   }
 }
 
-const retrieveMappings = async (http, settingsToRetrieve, dataSourceId) => {
-  const { body: mappings } = await retrieveSettings(
-    http,
-    'fields',
-    settingsToRetrieve,
-    dataSourceId
-  );
-  if (mappings) {
+// Type guard function
+function isHttpResponse<T>(object: any): object is HttpResponse<T> {
+  return object && typeof object === 'object' && 'body' in object;
+}
+
+const retrieveMappings = async (http: HttpSetup, settingsToRetrieve: any, dataSourceId: string) => {
+  const response = await retrieveSettings(http, 'fields', settingsToRetrieve, dataSourceId);
+
+  if (isHttpResponse(response) && response.body) {
+    const mappings = response.body;
     const maxMappingSize = Object.keys(mappings).length > 10 * 1024 * 1024;
-    let mappingsResponse;
+    let mappingsResponse: IndexMapping | IndexMappingOld | '{}';
     if (maxMappingSize) {
+      // eslint-disable-next-line no-console
       console.warn(
         `Mapping size is larger than 10MB (${
           Object.keys(mappings).length / 1024 / 1024
         } MB). Ignoring...`
       );
       mappingsResponse = '{}';
     } else {
-      mappingsResponse = mappings;
+      mappingsResponse = mappings as IndexMapping | IndexMappingOld;
     }
     loadMappings(mappingsResponse);
   }
 };
 
-const retrieveAliases = async (http, settingsToRetrieve, dataSourceId) => {
-  const { body: aliases } = await retrieveSettings(
-    http,
-    'indices',
-    settingsToRetrieve,
-    dataSourceId
-  );
+const retrieveAliases = async (http: HttpSetup, settingsToRetrieve: any, dataSourceId: string) => {
+  const response = await retrieveSettings(http, 'fields', settingsToRetrieve, dataSourceId);
```

Comment:
nit: do you think we should add some error handling in the file when we make some async requests?

Reply:
I have found ```sendRequestToOpenSearch``` that does error handling of requests. https://github.com/opensearch-project/OpenSearch-Dashboards/blob/2322c53d10664d68c7d2ee21a86a4e29d1919d9c/src/plugins/console/public/application/hooks/use_send_current_request_to_opensearch/send_request_to_opensearch.ts#L65-L195 Should we use this one instead? If it is the better way, I can try refactoring to use ```sendRequestToOpenSearch``` when doing refactor PR(for lodash and other legacy code).

Reply:
I mean something like try catch

```
try {
    const response = await retrieveSettings(http, 'fields', settingsToRetrieve, dataSourceId);

    if (isHttpResponse(response) && response.body) {
      const aliases = response.body as IndexAliases;
      loadAliases(aliases);
    }
  } catch (error) {
    console.error('Error retrieving aliases', error);
  }
```
I am okay with current change as well. not a blocker.